File: backend/src/main/java/br/com/unimedceara/portalcomunicacao/accesscontrol/domain/model/ColaboradorStatus.java
package br.com.unimedceara.portalcomunicacao.accesscontrol.domain.model;

/**
 * Status lógico do colaborador.
 */
public enum ColaboradorStatus {
    ACTIVE,
    INACTIVE;

    public static ColaboradorStatus fromFlag(String flag) {
        return "S".equalsIgnoreCase(flag) ? ACTIVE : INACTIVE;
    }

    public String toFlag() {
        return this == ACTIVE ? "S" : "N";
    }
}

File: frontend/src/composables/organization/colaborador-status.ts
import type { ColaboradorStatus } from "@/types/organization/colaborador.types";

export function resolveTargetColaboradorStatus(
  currentStatus: ColaboradorStatus
): ColaboradorStatus {
  return currentStatus === "ACTIVE" ? "INACTIVE" : "ACTIVE";
}

export function isColaboradorDeactivation(
  currentStatus: ColaboradorStatus
): boolean {
  return currentStatus === "ACTIVE";
}
